trace/trace.go

```diff
@@ -0,0 +1,122 @@
+package trace
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/sunfmin/reflectutils"
+	kitlog "github.com/theplant/appkit/log"
+	"github.com/theplant/prottp"
+	"google.golang.org/grpc"
+)
+
+// UnaryServerInterceptor new a server interceptor
+func UnaryServerInterceptor(log kitlog.Logger, traces []MethodTrace) grpc.UnaryServerInterceptor {
+	for i := range traces {
+		// MethodName: GetArticleById
+		// FullMethod: /theplant.ec.api.blog.BlogService/GetArticleById
+		traces[i].MethodName = fmt.Sprintf("/%s", traces[i].MethodName)
+	}
+
+	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (resp interface{}, err error) {
+		trace := findTrace(traces, info.FullMethod)
+		if trace == nil {
+			return
```

Comment:
This case doesn't call the handler now?

Reply:
Fixed it and add a test case https://github.com/theplant/prottp/pull/13/commits/64217a485116bb7d8135b409cff5715ca09af4c6